### PR TITLE
First Phase drafted

### PR DESCRIPTION
**Pull request by arnavjulka**
Please set the title of your pull request in the following format: `[Project id/Independent study/Issue fix/...]. project name`, eg., `E904. Integrating courses and assignments on Expertiza with Moodle`

We suggest you including the below information in your pull request:
- A description of the changes proposed in the pull request.
- @mentions of the person or team responsible for reviewing proposed changes.
---
About Expertiza Bot
- If you have any questions about comments given by the expertiza-bot [(example)](https://github.com/expertiza/expertiza/pull/1877#issuecomment-762412918), you could create a comment in your pull request with the format `/dispute [UUID1] [UUID2]` [(example)](https://github.com/expertiza/expertiza/pull/1877#issuecomment-762430057) then the professor and TAs will be notified.
- [Here is a video](https://tinyurl.com/internet-bots) about how to communicate to expertiza-bot.

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Course < ActiveRecord::Base
+  enum locale: {
+      en_US: 0,
+      hi_IN: 1
+  }
```
Comment (arnavjulka) [inline app/models/course.rb:5]: use same constant enum locale in all the occurrence, 
if not that atleast keep the enums with same values across the files

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 class User < ActiveRecord::Base
+  enum locale: {
+      no_pref: 0,
+      en_US: 1,
+      hi_IN: 2
+  }
```
Comment (arnavjulka) [inline app/models/user.rb:6]: reference to the previous comment

File: app/views/courses/_course.html.erb [hunk 3]
```diff
@@ -12,5 +12,7 @@
 <p><label for="course_info">Course Information</label><br/>
 <%= text_field 'course', 'info'  %></p>
 
+<p><label>Default Language</label><br/>
+  <%= select("course", "locale", [["English", "en_US"], ["Hindi", "hi_IN"]], {:selected => @course ? @course.locale : ["English", "en_US"]}) %></p>
```
Comment (arnavjulka) [inline app/views/courses/_course.html.erb]: same comment about locale enum

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -16,6 +16,31 @@ class ApplicationController < ActionController::Base
...
+      else
+        I18n.locale = I18n.default_locale
+      end
+    end
+  end
+
+  def get_locale_from_course
+    course = Course.find(params[:id])
+    course.locale
+  end
```
Comment (arnavjulka) [inline app/controllers/application_controller.rb]: put this function in the course model, and call it directly, send the argument there 

i believe it should look like this in the course model
```
def get_locale_from_course(id)
    course = Course.find(:id)
    course.locale
  end
```

and in this controller 
`get_locale_from_course(params[:id])`

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -16,6 +16,31 @@ class ApplicationController < ActionController::Base
...
+
+  def get_locale_from_assignment
+    assignment = Assignment.find(params[:id])
+    assignment.course.locale
+  end
```
Comment (arnavjulka) [inline app/controllers/application_controller.rb]: put this function in the Assignment table, and call it directly

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -16,6 +16,31 @@ class ApplicationController < ActionController::Base
   before_action :set_time_zone
   before_action :authorize
   before_action :filter_utf8
+  before_action :set_locale
+
+  def set_locale
+    if logged_in?
+      if @current_user.locale != "no_pref"
```
Comment (arnavjulka) [inline app/controllers/application_controller.rb:23]: shouldn't we be checking enums?

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (ashwindasr) [reply app/controllers/application_controller.rb]: Made the suggested modifications. Please check.

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (ashwindasr) [reply app/controllers/application_controller.rb]: Made the suggested modifications. Please check.

File: app/models/course.rb (hunk 1 above)
Comment (ashwindasr) [reply app/models/course.rb:5]: Need to discuss about updating migration files as well.

File: app/models/user.rb (hunk 2 above)
Comment (ashwindasr) [reply app/models/user.rb:6]: Need to discuss about updating migration files as well.

File: app/views/courses/_course.html.erb (hunk 3 above)
Comment (ashwindasr) [reply app/views/courses/_course.html.erb]: Fixed. Added new global variable in contoller.

**Pull request merged**